src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,97 @@
+package duke.storage;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.stream.Stream;
+
+import duke.exceptions.ProcessingException;
+import duke.exceptions.StartUpException;
+import duke.tasks.Task;
+
+/**
+ * The `Storage` class manages the interaction between the temporary storage (`TempStorage`)
+ * and local storage (`LocalStorage`) for saving, loading, updating, and performing operations
+ * on tasks in the Duke application.
+ */
+public class Storage {
+
+    private final LocalStorage local;
+    private final TempStorage temp;
+
+    /**
+     * Constructs a new `Storage` object with the specified file location.
+     *
+     * @param fileDirectory The file directory where save file is stored.
+     * @param fileName The name of the save file where the tasks are stored
+     * @throws StartUpException If an error occurs during the startup process, such as file access issues.
+     */
+    public Storage(String fileDirectory, String fileName) throws StartUpException {
+        temp = new TempStorage();
+
+        try {
+            local = new LocalStorage(fileDirectory, fileName);
+            if (local.hasCreatedNewFile()) {
+                saveFileNotFound();
+            } else {
+                saveFileFound();
+                ArrayList<String> loadList = local.load();
+                temp.load(loadList); //Load from local into temp
+            }
+
```

Comment:
Maybe there is an extra empty line to delete? I noticed the same issue in several other places too.

Reply:
Yep, thanks for catching that!